tsdb/engine/wal/wal.go

```diff
@@ -238,9 +238,8 @@ func (l *Log) WritePoints(points []tsdb.Point, fields map[string]*tsdb.Measureme
 
 	// persist the raw point data
 	l.mu.RLock()
-	defer l.mu.RUnlock()
-
 	partitionsToWrite := l.pointsToPartitions(points)
+	l.mu.RUnlock()
```

Comment:
Any point adding a comment.....`// persist the raw point data, holding the lock for as short as possible".

I am a little concerned future developers would revert to defer. We're adding some important changes, but it may not be obvious why in the future.